src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -815,6 +815,54 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         return builder->CreateCall(fn, {str, idx1, idx2});
     }
 
+    llvm::Value* lfortran_float_to_str(llvm::Value* number)
+    {
+        std::string func_name = "_lfortran_float_to_str";
+        llvm::Function *fn = module->getFunction(func_name);
+        if (!fn) {
+            llvm::FunctionType *function_type = llvm::FunctionType::get(
+                    character_type, {
+                        llvm::Type::getFloatTy(context)
+                    }, false);
+            fn = llvm::Function::Create(function_type,
+                    llvm::Function::ExternalLinkage, func_name, *module);
+        }
+        llvm::Value* res = builder->CreateCall(fn, {number});
+        return res;
+    }
+
+    llvm::Value* lfortran_int_to_str(llvm::Value* number)
+    {
+        std::string func_name = "_lfortran_int_to_str";
+        llvm::Function *fn = module->getFunction(func_name);
+        if(!fn) {
+            llvm::FunctionType *function_type = llvm::FunctionType::get(
+                character_type, {
+                    llvm::Type::getInt32Ty(context)
+                }, false);
+            fn = llvm::Function::Create(function_type,
+                    llvm::Function::ExternalLinkage, func_name, *module);
+        }
+        llvm::Value* res = builder->CreateCall(fn, {number});
+        return res;
+    }
+
+    llvm::Value* lfortran_bool_to_str(llvm::Value* number)
+    {
+        std::string func_name = "_lfortran_bool_to_str";
+        llvm::Function *fn = module->getFunction(func_name);
+        if(!fn) {
+            llvm::FunctionType *function_type = llvm::FunctionType::get(
+                character_type, {
+                    llvm::Type::getInt1Ty(context)
+                }, false);
+            fn = llvm::Function::Create(function_type,
+                    llvm::Function::ExternalLinkage, func_name, *module);
+        }
+        llvm::Value* res = builder->CreateCall(fn, {number});
+        return res;
+    }
+
```

Comment:
Common logic of `_lfortran_int_to_str`, `_lfortran_float_to_str`, `_lfortran_bool_to_str` can be extracted out in a common utility function say, `_lfortran_type_to_str`. The interface of this common function can look like,

```cpp
llvm::Value* _lfortran_type_to_str(llvm::Value* number, llvm::Type* value_type, std::string type) {
        std::string func_name = "_lfortran_" + type + "_to_str";
         llvm::Function *fn = module->getFunction(func_name);
         if(!fn) {
             llvm::FunctionType *function_type = llvm::FunctionType::get(
                 character_type, {
                     value_type
                 }, false);
             fn = llvm::Function::Create(function_type,
                     llvm::Function::ExternalLinkage, func_name, *module);
         }
         llvm::Value* res = builder->CreateCall(fn, {number});
         return res;
     }
}
```